### PR TITLE
buf_format(): parse format string only once

### PR DESCRIPTION
**Pull request by p371jah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/util/buf.c [hunk 1]
```diff
@@ -81,97 +81,143 @@ static bool buf_resize(RzBuffer *b, ut64 newsize) {
...
+				while (nbytes != 0) {
+					size_t block_sz = RZ_MIN(sizeof(tmp), nbytes);
+					if (rz_buf_read(src, tmp, block_sz) != block_sz) {
+						goto err_exit;
+					}
+					if (rz_buf_write(dst, tmp, block_sz) != block_sz) {
+						goto err_exit;
+					}
+					res += block_sz;
+					nbytes -= block_sz;
```
Comment (p371jah) [inline librz/util/buf.c]: PR also contains small optimization of reading chars

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
```
 librz/util/buf.c:158:20: error: code should be clang-formatted [-Wclang-format-violations]
                rz_vector_foreach (&tokens, tok) {
                                 ^
```

**Comment by wargio** [comment]
@pelijah what is that component used for ? auto-detecting strings that are used as buffer within rizin?

**Review by wargio — APPROVED** [review]
OH nice fix!! i didn't notice by what this was used for.

**Pull request merged**